### PR TITLE
Add SslClient and SslServer implementations for security-framework

### PR DESCRIPTION
**Pull request by sfackler**

### CODE REVIEW

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: 4bf4accc5a4b31724f2858858962bf741dde025a
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 6cf43c1bd0b72349de23530dddef0aadca37557c
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by jdm** [comment]
💃

File: src/net.rs [hunk 1]
```diff
@@ -593,14 +593,17 @@ impl<S: SslClient> NetworkConnector for HttpsConnector<S> {
...
+#[cfg(all(not(feature = "openssl"), not(feature = "security-framework")))]
 #[doc(hidden)]
 pub type DefaultConnector = HttpConnector;
 
 #[cfg(feature = "openssl")]
 #[doc(hidden)]
 pub type DefaultConnector = HttpsConnector<self::openssl::Openssl>;
 
+#[cfg(all(feature = "security-framework", not(feature = "openssl")))]
+pub type DefaultConnector = HttpsConnector<self::security_framework::ClientWrapper>;
```
Comment (seanmonstar) [inline src/net.rs]: Openssl feature would still be enabled to provide the secure feature of cookies. So if you enable both, you'd be using the Openssl connector, right?

**Comment by seanmonstar** [comment]
\o/

File: src/net.rs (hunk 1 above)
Comment (sfackler) [inline src/net.rs]: If you're using that feature, yeah, though that's a thing that we can fix on the cookie side. But it seems a bit unlikely that you'd be enabling that feature anyway if you're using the client side of hyper.

**Pull request merged**